rotkehlchen/globaldb/updates/manager.py

```diff
@@ -837,3 +687,52 @@ def _retrieve_update_files(
             }
 
         return updates
+
+    def apply_pending_compatible_updates(self) -> None:
+        """Apply any pending asset updates that are compatible with the current DB version
+        before upgrading to target_db_version.
+
+        This ensures we don't miss any asset updates that would become incompatible
+        after the DB upgrade.
+        """
+        max_compatible_version = None
+        info_json = self._get_remote_info_json()
+        latest_assets_version = info_json['latest']
```

Comment:
This entire function seems to not check for any key errors. How do you guarantee all keys exist? Does it not depend on a remote call?